File: src/main/java/uk/gov/justice/hmpps/casenotes/repository/CaseNoteTypeRepository.java
package uk.gov.justice.hmpps.casenotes.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import uk.gov.justice.hmpps.casenotes.model.SensitiveCaseNoteType;

@Repository
public interface CaseNoteTypeRepository extends JpaRepository<SensitiveCaseNoteType, Long> {

    SensitiveCaseNoteType findSensitiveCaseNoteTypeByParentType_TypeAndType(String parentType, String type);
}

File: src/main/java/uk/gov/justice/hmpps/casenotes/repository/CaseNoteTypeRepository.kt
package uk.gov.justice.hmpps.casenotes.repository

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.stereotype.Repository
import uk.gov.justice.hmpps.casenotes.model.CaseNoteType
import java.util.Optional

@Repository
interface CaseNoteTypeRepository : JpaRepository<CaseNoteType, Long> {
  @Query(
    """
        select t from CaseNoteType t
        join fetch t.parentType pt
        where pt.type = :parentType and t.type = :type
    """,
  )
  fun findByParentTypeAndType(parentType: String, type: String): Optional<CaseNoteType>
}
